### PR TITLE
Noloan - Rework SmsMessages to DbMessages

### PR DESCRIPTION
**Pull request by gulilo**
Removed Inbox from `SmsMessages`
renamed to `DbMessages`
change `DbMessages` functionlity to be the main class in the `Firebase feature`

### CODE REVIEW

File: projects/noloan/admin/src/main/java/hasadna/noloan/admin/app/SpamRecyclerAdapter.java [hunk 1]
```diff
@@ -13,25 +13,23 @@
...
 
 public class SpamRecyclerAdapter
     extends RecyclerView.Adapter<SpamRecyclerAdapter.RecyclerViewHolder> {
 
-  SmsMessages DbMessages;
 
   public SpamRecyclerAdapter() {
 
-    DbMessages = SmsMessages.get();
+    DbMessages dbMessages = hasadna.noloan.common.DbMessages.getInstance();
```
Comment (oferb) [inline projects/noloan/admin/src/main/java/hasadna/noloan/admin/app/SpamRecyclerAdapter.java]: No need for the package name `hasadna.noloan.common`

File: projects/noloan/admin/src/main/java/hasadna/noloan/admin/app/SplashScreenActivity.java [hunk 2]
```diff
@@ -93,14 +93,11 @@ private void startNextActivity() {
     protected Object doInBackground(Object[] objects) {
       checkPermissions();
       Task spam = new FirestoreClient().StartListeningToMessages().getTask();
-      Task Suggestions = new FirestoreClient().StartListeningToMessages().getTask();
-
       Task<Boolean> permissions = permissionTask.getTask();
       try {
 
         Tasks.await(permissions);
         Tasks.await(spam);
```
Comment (oferb) [inline projects/noloan/admin/src/main/java/hasadna/noloan/admin/app/SplashScreenActivity.java]: I guess you can rename `spam` to `messages`?

File: projects/noloan/admin/src/main/java/hasadna/noloan/admin/app/SuggetionRecyclerAdapter.java [hunk 3]
```diff
@@ -81,15 +82,13 @@ public void bind(SmsMessage sms) {
       receivedAt.setText(sms.getReceivedAt());
       buttonAccept.setOnClickListener(
           view -> {
-            FirestoreClient client = new FirestoreClient();
-            client.writeMessage(sms);
```
Comment (oferb) [inline projects/noloan/admin/src/main/java/hasadna/noloan/admin/app/SuggetionRecyclerAdapter.java]: Why did we add and then delete?

File: projects/noloan/app/src/main/java/hasadna/noloan/InboxRecyclerAdapter.java [hunk 4]
```diff
@@ -75,41 +79,25 @@ public void bind(SmsMessage sms) {
...
-            itemView
-                .getResources()
-                .getString(
-                    R.string.list_item_textView_spam_counter,
-                    SmsMessages.get()
-                        .getDbMessages()
-                        .get(SmsMessages.get().searchDbMessage(sms))
-                        .getSuggestersCount()));
+      DbMessages dbMessages = DbMessages.getInstance();
+      int index = DbMessages.findSms(sms, dbMessages.getMessages());
```
Comment (oferb) [inline projects/noloan/app/src/main/java/hasadna/noloan/InboxRecyclerAdapter.java]: You can create static methods like `getMessages()` that call `DbMessages.getInstance().getMessages()` so you don't have to use `DbMessages dbMessages = DbMessages.getInstance();` every time, and instead just do `DbMessages.getMessages()` for example.

File: projects/noloan/app/src/main/java/hasadna/noloan/SpamRecyclerAdapter.java [hunk 5]
```diff
@@ -133,16 +133,17 @@ public void bind(SmsMessage sms) {
...
 
       buttonAddSuggestion.setOnClickListener(
           view -> {
-            SmsMessages.get().suggestMessage(sms);
+            DbMessages.getInstance().addMessage(sms);
           });
     }
 
     // Displays the "Undo suggestion" button, in case user had suggested this message.
+    //TODO for consistesy, remove this function
```
Comment (oferb) [inline projects/noloan/app/src/main/java/hasadna/noloan/SpamRecyclerAdapter.java]: `//TODO for consistesy->// TODO for consistency`

File: projects/noloan/common/src/main/java/hasadna/noloan/common/DbMessages.java [hunk 6]
```diff
@@ -0,0 +1,174 @@
+package hasadna.noloan.common;
+
+import android.util.Log;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import hasadna.noloan.protobuf.SmsProto.SmsMessage;
+
+// Simple singleton to hold smsMessages from the db, smsMessages from the inbox
```
Comment (oferb) [inline projects/noloan/common/src/main/java/hasadna/noloan/common/DbMessages.java]: `smsMessages->SMS messages`

Comment (oferb) [inline projects/noloan/common/src/main/java/hasadna/noloan/common/DbMessages.java]: `smsMessages from the inbox` - where is this stored?

File: projects/noloan/common/src/main/java/hasadna/noloan/common/DbMessages.java [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+    void messageAdded(SmsMessage newMessage);
+
+    void messageModified(int index);
+
+    // SmsMessage parameter is passed for cases the index already had been removed from the db list
+    void messageRemoved(int index, SmsMessage smsMessage);
+
+  }
+
+  // static function to search a given list
```
Comment (oferb) [inline projects/noloan/common/src/main/java/hasadna/noloan/common/DbMessages.java]: No need to repeat the code in the comment, so I'd remove `static function`.
I'd rephrase to:
`// Searches in the given list`

File: projects/noloan/admin/src/main/java/hasadna/noloan/admin/app/SuggetionRecyclerAdapter.java (hunk 3 above)
Comment (gulilo) [reply projects/noloan/admin/src/main/java/hasadna/noloan/admin/app/SuggetionRecyclerAdapter.java]: We added to the spam list and deleted from the suggested list. 
Told you admin is not working for a while now

File: projects/noloan/common/src/main/java/hasadna/noloan/common/DbMessages.java (hunk 6 above)
Comment (gulilo) [reply projects/noloan/common/src/main/java/hasadna/noloan/common/DbMessages.java]: forgot to change that description
